mount/mount_linux.go

```diff
@@ -42,16 +44,35 @@ func (m *Mount) Mount(target string) error {
 }
 
 // Unmount the provided mount path with the flags
-func Unmount(mount string, flags int) error {
-	return unix.Unmount(mount, flags)
+func Unmount(target string, flags int) error {
+	if err := unmount(target, flags); err != nil && err != unix.EINVAL {
+		return err
+	}
+	return nil
+}
+
+func unmount(target string, flags int) error {
```

Comment:
This doesn't actually fix the issue. It just reduce the chance, right?

Should we still continue `os.RemoveAll` when unmount fails?

Reply:
We have had this type of logic in the past for unmounting,  EBUSY/EAGAIN means "try again" so its still an other all valid fix for the type of issue.

@dmcgowan suggested that we change RemoveAll to Remove so that we either leak a temp dir if it fails but not RM snapshot data.

Reply:
SGTM